File: arrays.rb
num = ARGV[0].to_i

array = []
start_time = Time.now
num.times do |i|
  array << i
end
end_time = Time.now

size = array.count.to_s.reverse.gsub(/(\d{3})(?=\d)/, '\\1,').reverse
puts "Done.  Array size is #{size}"

elapsed = end_time - start_time
na, ms = (elapsed * 1000).divmod(1000)
mm, ss = elapsed.divmod(60)
hh, mm = mm.divmod(60)
dd, hh = hh.divmod(24)

puts "Time: %d hours, %d minutes, %d seconds, %05.2f milliseconds" % [hh, mm, ss, ms]

# Done.  Array is 1,000,000
# Time: 00.00 hours, 00.00 minutes, 00 seconds, 132.54 milliseconds

# Done.  Array is 10,000,000
# Time: 0 hours, 0 minutes, 1 seconds, 246.86 milliseconds

# Done.  Array is 100,000,000
# Time: 0 hours, 0 minutes, 13 seconds, 73.54 milliseconds

# Done.  Array is 1,000,000,000
# Time: 0 hours, 4 minutes, 44 seconds, 226.27 milliseconds

File: arrays.py
# come on python, save us
import sys, time, datetime
import locale

locale.setlocale(locale.LC_ALL, 'en_US')

# another way to do it
# datetime.timedelta(milliseconds=elapsed)
# but it's not equivalent
def prntime(ms): 
	s=ms/1000
	na,milli = divmod((ms * 1000), 1000)
	m,s=divmod(s,60) 
	h,m=divmod(m,60) 
	return h,m,s,milli

try:
	sys.argv[1]
except IndexError:
	print "hey, give me a number for array size like 1000000"
	exit(1)

num = int(sys.argv[1])

array = []
start_time = datetime.datetime.now()
for i in range(0, num):
	array.append(i)

end_time = datetime.datetime.now()

size = locale.format("%d", len(array), grouping=True)
print "Done. Array size is %s" % size

elapsed = (end_time - start_time).total_seconds() * 1000

print 'Time: %d hours, %d minutes, %d seconds, %d milliseconds' % prntime(elapsed) 


# Done. Array size is 1,000,000
# Time: 0 hours, 0 minutes, 0 seconds, 993 milliseconds

# Done. Array size is 10,000,000
# Time: 0 hours, 0 minutes, 2 seconds, 932 milliseconds

# Done. Array size is 100,000,000
# Time: 0 hours, 0 minutes, 25 seconds, 25 milliseconds

# managed to use 28gb of memory
